src/openforms/tests/e2e/test_input_validation.py

```diff
@@ -292,3 +317,223 @@ def test_max_date_fixed_value(self):
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            api_value=False,
+            expected_ui_error="Het verplichte veld Required checkbox is niet ingevuld.",
+        )
+
+
+class SingleCurrencyTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required currency is niet ingevuld.",
+        )
+
+    def test_min_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"min": 10.7},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=2,
+            expected_ui_error="De waarde moet 10.7 of groter zijn.",
+        )
+
+    def test_max_value(self):
+        component: Component = {
+            "type": "currency",
+            "key": "requiredCurrency",
+            "label": "Required currency",
+            "validate": {"max": 15},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=50,
+            expected_ui_error="De waarde moet 15 of kleiner zijn.",
+        )
+
+
+class SingleMapTests(ValidationsTestCase):
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        await page.wait_for_selector(
+            f".openforms-leaflet-map, [aria-label='{label}']", state="visible"
+        )
+
+    def test_required_field(self):
+        component: Component = {
+            "type": "map",
+            "key": "requiredMap",
+            "label": "Required map",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required map is niet ingevuld.",
+        )
+
+
+class SinglePostcodeTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "postcode",
+            "key": "requiredPostcode",
+            "label": "Required postcode",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required postcode is niet ingevuld.",
+        )
+
+
+class SingleSignatureTests(ValidationsTestCase):
+
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector(f"[aria-label='{label}']", state="visible")
```

Comment:
seems to also need the await: https://playwright.dev/python/docs/api/class-elementhandle#element-handle-wait-for-selector

```suggestion
        await page.wait_for_selector(f"[aria-label='{label}']", state="visible")
```